core/server/worker/src/main/java/alluxio/worker/page/PagedBlockStore.java

```diff
@@ -116,22 +116,23 @@ public void unpinBlock(long id) {
   @Override
   public void commitBlock(long sessionId, long blockId, boolean pinOnCreate)
       throws IOException {
-    // TODO(bowen): implement actual committing and replace placeholder values
-    BlockStoreLocation dummyLoc = new BlockStoreLocation(DEFAULT_TIER, 1);
+    PageStoreDir pageStoreDir =
+        mPageMetaStore.allocate(String.valueOf(blockId), 0);
+    pageStoreDir.commit(String.valueOf(blockId));
+    BlockStoreLocation blockLocation =
+        new BlockStoreLocation(DEFAULT_TIER, mPageMetaStore.getStoreDirs().indexOf(pageStoreDir));
```

Comment:
In #[16024](https://github.com/Alluxio/alluxio/pull/16024), I make `PagedBlockStoreDir` implement both `PageStoreDir` and `StorageDir` at the same time, which has a `getDirIndex` method.